octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/DescriptionComposite.java

```diff
@@ -1,69 +1,160 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package com.hpe.octane.ideplugins.eclipse.ui.entitydetail.field;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.browser.Browser;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.util.IOUtils;
+import com.hpe.adm.nga.sdk.authentication.SimpleUserAuthentication;
+import com.hpe.adm.nga.sdk.model.EntityModel;
+import com.hpe.adm.octane.ideplugins.services.util.Util;
+import com.hpe.octane.ideplugins.eclipse.Activator;
+import com.hpe.octane.ideplugins.eclipse.ui.entitydetail.model.EntityModelWrapper;
+import com.hpe.octane.ideplugins.eclipse.ui.util.ClientLoginCookie;
+import com.hpe.octane.ideplugins.eclipse.ui.util.LinkInterceptListener;
+import com.hpe.octane.ideplugins.eclipse.ui.util.PropagateScrollBrowserFactory;
+import com.hpe.octane.ideplugins.eclipse.ui.util.StackLayoutComposite;
+import com.hpe.octane.ideplugins.eclipse.ui.util.resource.PlatformResourcesManager;
+import com.hpe.octane.ideplugins.eclipse.util.EntityFieldsConstants;
+
+public class DescriptionComposite extends Composite {
+
+    private PropagateScrollBrowserFactory factory = new PropagateScrollBrowserFactory();
+    private Color foregroundColor = PlatformResourcesManager.getPlatformForegroundColor();
+    private Color backgroundColor = PlatformResourcesManager.getPlatformBackgroundColor();
+    private Browser browserDescHtml;
+    private File octanePhotosDir;
+
+    public DescriptionComposite(Composite parent, int style) {
+        super(parent, style);
+        setLayout(new FillLayout(SWT.HORIZONTAL));
+        StackLayoutComposite stackLayoutComposite = new StackLayoutComposite(this, SWT.NONE);
+        browserDescHtml = factory.createBrowser(stackLayoutComposite, SWT.NONE);
+        browserDescHtml.addLocationListener(new LinkInterceptListener());
+        stackLayoutComposite.showControl(browserDescHtml);
+    }
+
+    public void setEntityModel(EntityModelWrapper entityModelWrapper) {
+        browserDescHtml.setText(getBrowserText(entityModelWrapper.getReadOnlyEntityModel()));
+    }
+
+    public String getBrowserText(EntityModel entityModel) {
```

Comment:
@vladbenchea getBrowserText seems to only be used above in setEntityModel. Is it necessary to make it "public"?

Reply:
No, it should be private.